glue/core/layer_artist.py

```diff
@@ -228,6 +228,12 @@ def _check_subset_state_changed(self):
             self._changed = True
             self._state = state
 
+    def update_component_limits(self, components_changed):
```

Comment:
Can you give a real life example of what this method would do? I wonder if this should be a more general method such as ``on_component_change`` or something similar that could be more versatile?

Reply:
Sure -- in glue genes we have a SyncComponent which is a component that is added to a dataset when a subset is created and is updated as that subset is updated. For instance, we can calculate gene expression levels in a spatial dataset for a given subset of genes; as we change the subset of genes, we update the values in the SyncComponent on the spatial dataset, but we need to prompt the cmap_lim_helpers so that they display the full range of values in the new component. 

Thus, a Viewer can add something like this:

```
@defer_draw
    def update_component_limits(self, components_changed):
        for limit_helper in [self.state.cmap_lim_helper]:
            if limit_helper.attribute in components_changed:
                limit_helper.update_values(force=True)
        self.redraw()
```

And the user experience is nice because changing the subset automatically scales the color to show the range of values. 
![Screen Recording 2023-09-21 at 5 10 13 PM](https://github.com/glue-viz/glue/assets/3639698/37a596bc-36d8-459e-8e71-687c7c250511)

This could definitely be called something more general; I just had the one example and I thought a more specific name would be more readable in that case.

Reply:
Perhaps the main 'hook' that gets called could be called with a more general name, and in your case it could then call from there ``update_component_limits``?

Reply:
Yeah, sounds good. I've updated to this approach.